src/views/docs-view/utils/get-custom-layout.ts

```diff
@@ -0,0 +1,28 @@
+import { RootDocsPath } from 'types/products'
+
+/**
+ * TODO: this logic is temporary to showcase how layouts can work.
+ * Going forward this would be defined in frontmatter,
+ * we'd pass frontmatter.layout directly rather than constructing this.
+ *
+ * Task: https://app.asana.com/0/1202097197789424/1202850056121889/f
+ */
+export function getCustomLayout({
+	currentRootDocsPath,
+	frontMatter,
+	pathParts,
+}: {
+	currentRootDocsPath: RootDocsPath
+	frontMatter: Record<string, unknown>
+	pathParts: string[]
+}) {
+	const isLanding = pathParts.length === 0
+	const customLayout = isLanding
+		? {
+				name: 'docs-root-landing',
+				subtitle:
+					currentRootDocsPath.description ?? frontMatter.description ?? null,
+		  }
+		: null
+	return customLayout
+}
```

Comment:
💭 optional suggestion: if we wanna merge this without production impact, we could put it behind a flag for now at this point! We can refine and iterate on the empty state as a follow-up this way.

```suggestion
export function getCustomLayout({
	currentRootDocsPath,
	frontMatter,
	pathParts,
}: {
	currentRootDocsPath: RootDocsPath
	frontMatter: Record<string, unknown>
	pathParts: string[]
}) {
  if (!__config.flags.enable_custom_layouts) {
    return null
  }
	const isLanding = pathParts.length === 0
	const customLayout = isLanding
		? {
				name: 'docs-root-landing',
				subtitle:
					currentRootDocsPath.description ?? frontMatter.description ?? null,
		  }
		: null
	return customLayout
}
```

Reply:
Sweet, I really dig this 💖  In this specific case, I think it feels best to merge in with the empty state we have so far - intent being to avoid getting into a state where we have to kind of "clean up" feature flags later (that would feel a bit like papercut-ish debt!). This is definitely based on the recent design approval below though too - thanks @braydenlove !

I'll definitely keep feature flags in mind for future work 👍 💯